connectors/services/job_scheduling.py

```diff
@@ -28,6 +31,44 @@
 from connectors.source import get_source_klass
 
 
+class LastSyncTimeMethods(Enum):
```

Comment:
Instead of introducing a new class. WDYT if we introduce 2 new methods in `Connector` class?
```python
def last_sync_scheduled_at_by_job_type(job_type):
    match job_type:
        case JobType.ACCESS_CONTROL:
            return self.last_access_control_sync_scheduled_at
        case JobType.INCREMENTAL:
            return self.last_incremental_sync_scheduled_at
        case JobType.FULL:
            return self.last_full_sync_scheduled_at
        case _:
            raise ValueError(f"Unknown job type: {job_type}")

async def update_last_sync_scheduled_at_by_job_type(job_type, new_ts):
    match job_type:
        case JobType.ACCESS_CONTROL:
            await self._update_datetime("last_access_control_sync_scheduled_at", new_ts)
        case JobType.INCREMENTAL:
            await self._update_datetime("last_incremental_sync_scheduled_at", new_ts)
        case JobType.FULL:
            await self._update_datetime("last_sync_scheduled_at", new_ts)
        case _:
            raise ValueError(f"Unknown job type: {job_type}")
```

Reply:
Makes sense! Increases the cohesion and removes the "magic" lookups. Will adjust it 👍